types.go

```diff
@@ -53,9 +53,12 @@ func (date NumericDate) MarshalJSON() (b []byte, err error) {
 	if TimePrecision < time.Second {
 		prec = int(math.Log10(float64(time.Second) / float64(TimePrecision)))
 	}
-	f := float64(date.Truncate(TimePrecision).UnixNano()) / float64(time.Second)
+	trancatedDate := date.Truncate(TimePrecision)
+	whole := strconv.FormatInt(trancatedDate.Unix(), 10)
+	decimal := strconv.FormatFloat(float64(trancatedDate.Nanosecond())/float64(time.Second), 'f', prec, 64)
+	f := append([]byte(whole), []byte(decimal)[1:]...)
```

Comment:
I think a small comment to the whole block would be good, in order to understand what is happening here.

Reply:
Done